src/neo/Wallets/Wallet.cs

```diff
@@ -593,6 +593,7 @@ public long CalculateNetworkFee(DataCache snapshot, Transaction tx)
         /// <returns><see langword="true"/> if the signature is successfully added to the context; otherwise, <see langword="false"/>.</returns>
         public bool Sign(ContractParametersContext context)
         {
+            if (context.Magic != ProtocolSettings.Magic) return false;
```

Comment:
The idea it's avoid this, isn't it?

Reply:
I think we need to check whether wallet has the same magic as ContractParametersContext, as this is its designed functionality

Reply:
I think that we should remove this check, what do you think @neo-project/core ?